src/PostgREST/App.hs

```diff
@@ -141,27 +144,29 @@ postgrestResponse appState conf@AppConfig{..} maybeSchemaCache pgVer authResult@
 
   body <- lift $ Wai.strictRequestBody req
 
-  let jwtTime = if configServerTimingEnabled then Auth.getJwtDur req else Nothing
+-- the preference metrics=timings cant be used before it is parsed, hence
+-- parseTime will be calculated for all requests
+  (parseTime, apiReq@ApiRequest{..}) <- withTiming True $ liftEither . mapLeft Error.ApiRequestError $ ApiRequest.userApiRequest conf req body sCache
```

Comment:
Ah, so it's not possible to disable ALL timings with `Prefer: metrics=timings`, without parsing the headers first. So it's not a perfect replacement of `server-timings-enabled = false`, since some timings will be calculated even if the header is not sent: the "jwt" and "parse" timings as far as I can see.

Not sure if it's OK to always calculate these timings, I believe it will still affect the performance as mentioned in the issue. cc: @steve-chavez

Reply:
@steve-chavez Any update on this?

Reply:
Hm, so here we're forcing to always calculate the time for parsing? That would defeat the whole purpose of https://github.com/PostgREST/postgrest/issues/3410

But I understand that we first need to parse to get the header. I guess we would need to do this in a step separate from `userApiRequest`, but that's messy.

Reply:
So far the `parse` and `response` timings ([ref](https://docs.postgrest.org/en/v12/references/observability.html#server-timing-header)) have not been that useful because they're always fast. 

Another option could be to omit the `parse` timing when doing `Prefer: metrics=timings` or we could just remove support for the above mentioned. Then we wouldn't have this problem.